app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -733,19 +743,25 @@ export const simulatedCall = async (ethereumClientService: EthereumClientService
 	}
 }
 
-const getSignedMessagesWithFakeSigner = (simulationState: SimulationState | undefined) => {
-	return simulationState === undefined ? [] : simulationState.signedMessages.map((x) => ({ fakeSignedFor: x.fakeSignedFor, originalRequestParameters: x.originalRequestParameters }))
+const simulateTransactionsOnTopOfSimulationInput = async (ethereumClientService: EthereumClientService, requestAbortController: AbortController | undefined, simulationStateInput: SimulationStateInput, transactions: EthereumUnsignedTransaction[], extraOverrides: StateOverrides = {}) => {
+	if (transactions.length === 0) return []
+	const signedTransactions = transactions.map((transaction) => mockSignTransaction(transaction))
+	const simulationStateInputWithNewTransactions = { blocks: [...simulationStateInput.blocks, {
+		transactions: [...signedTransactions.map((signedTransaction) => ({ signedTransaction: signedTransaction }) )],
+		stateOverrides: extraOverrides,
+		signedMessages: [],
+		timeIncreaseDelta: 1n, //TODO, change to 0 when geth supports same timestamp simulaions
```

Comment:
My preference is that we match chain blocktime.  Can we figure this out by looking at most recent two blocks and diffing the timestamp (when chain is changed) and using that as the block timestamp?  I suppose this will break if there is a missed slot on Ethereum.  🤔 

I would also accept `1n` for timestamp increase.  I don't think we should ever be doing blocks with 0 time between them.